content/blog/posts/2022-07-08-go-serverless-functions-with-ember-and-sst-on-aws.md

```diff
@@ -0,0 +1,200 @@
+    environment,
+    rootURL: '/',
+    locationType: 'history',
+    EmberENV: {
+      FEATURES: {},
+      EXTEND_PROTOTYPES: {
+        Date: false,
+      },
+    },
+
+    APP: {},
+  };
+
+  if (environment === 'development') {
+    ENV.APP.EMBER_APP_API_URL = process.env.EMBER_APP_API_URL_DEV;
+  }
+
+  if (environment === 'test') {
+    ENV.locationType = 'none';
+
+    // keep test console output quieter
+    ENV.APP.LOG_ACTIVE_GENERATION = false;
+    ENV.APP.LOG_VIEW_LOOKUPS = false;
+
+    ENV.APP.rootElement = '#ember-testing';
+    ENV.APP.autoboot = false;
+  }
+
+  if (environment === 'production') {
+    ENV.APP.EMBER_APP_API_URL = process.env.EMBER_APP_API_URL_PROD;
+  }
+
+  return ENV;
+};
+```
+
+And finally, we'll tweak the serverless function to respond with JSON rather than a string:
+
+```go
+// ./services/functions/lambda/main.go
+package main
+
+import (
+	"encoding/json"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func Handler(request events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
+	data := map[string]string{"message": "Hello, World!"}
+	json, err := json.Marshal(data)
+
+	return events.APIGatewayProxyResponse{
+		Body:       string(json),
+		StatusCode: 200,
+	}, err
+}
+
+func main() {
+	lambda.Start(Handler)
+}
+```
+
+With that wired up you can fire up the Ember app as you normally would and check in the browser that the serverless function is returning the data we expected:
+
+![Browser showing the Go serverless function response](/img/blog/serverlesss-functions-direct-to-aws-with-ember/browser-showing-go-serverless-response.png)
+
+And there is our Ember app with a Go-based serverless function that is deployable to AWS, and we only need to take one more step to deploy it!
+
+## Deployment to AWS
+
+If we run `npx sst deploy --stage prod`, SST will handle all the details of deploying the app. Again, this will take several minutes to completely deploy.
+
+Once it's done we'll need to set the prod `ApiEndpoint` URL provided in the output of the deploy logs as an environment variable in the Ember app and re-deploy using the same command. Once that deploys you've got an Ember app with a Go lang serverless function deployed to AWS!
+
+![Browser showing the CloudFront distribution deployment](/img/blog/serverlesss-functions-direct-to-aws-with-ember/cloudfront-distribution.png)
+
+and ...
+
+![Browser showing the Go serverless function response in the prod deployment](/img/blog/serverlesss-functions-direct-to-aws-with-ember/browser-showing-go-serverless-response-prod-deploy.png)
+
+Now, this is not the only way you can deploy serverless functions and an Ember application directly to AWS, but it is a convenient way to get that done.
+
+In future posts on this topic, we might look at alternative means for deployment that require a little more configuration, but which may also provide more control over the deployment and architecture, so stay tuned!
```

Comment:
In future posts on this topic, we might look at alternative means for deployment that requireS a little more configuration, but ~which may~ also provideS more control over the deployment and architecture, so stay tuned!

correction for tense